test/entrypoint/srv_ddl.lua

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env tarantool
+
+require('strict').on()
+_G.is_initialized = function() return false end
+
+local log = require('log')
+local errors = require('errors')
+local cartridge = require('cartridge')
+local ddl = require('ddl')
+
+package.preload['customers-storage'] = function()
+    return {
+        role_name = 'customers-storage',
+        init = function()
+            local engine = os.getenv('ENGINE') or 'memtx'
+            local schema = {
+                spaces = {
+                    customers = {
+                        engine = engine,
+                        is_local = true,
+                        temporary = false,
+                        format = {
+                            {name = 'id', is_nullable = false, type = 'unsigned'},
+                            {name = 'bucket_id', is_nullable = false, type = 'unsigned'},
+                            {name = 'name', is_nullable = false, type = 'string'},
+                            {name = 'age', is_nullable = false, type = 'number'},
+                        },
+                        indexes = {{
+                            name = 'id',
+                            type = 'TREE',
+                            unique = true,
+                            parts = {
+                                {path = 'id', is_nullable = false, type = 'unsigned'},
+                            },
+                        }, {
+                            name = 'bucket_id',
+                            type = 'TREE',
+                            unique = false,
+                            parts = {
+                                {path = 'bucket_id', is_nullable = false, type = 'unsigned'},
+                            }
+                        }},
+                        sharding_key = {'id', 'name'},
+                    },
+                }
+            }
+
+            rawset(_G, 'add_extra_field', function(name)
+                local new_format = box.space.developers:format()
+                table.insert(new_format, {name = name, type = 'any', is_nullable = true})
+                box.space.developers:format(new_format)
+            end)
+
+            if not box.cfg.read_only then
```

Comment:
It's better to use box.info.ro. `box.cfg.read_only` is that user passed in box.cfg{} but even `box.cfg{read_only = false}` specified instance could be read-only (e.g. in orphan mode)

Reply:
replaced with `box.info.ro`